src/queries/auth.tsx

```diff
@@ -0,0 +1,90 @@
+import { UserAttributes } from '@supabase/supabase-js';
+import supabase from '../utils/supabase';
+import { AuthDispatch, useAuthReducer } from '../utils/AuthContext';
+
+export const signInWithEmail = async (
+  dispatch: AuthDispatch,
+  email: string,
+  password: string,
+) => {
+  dispatch({ type: 'LOADING' });
+  const value = await supabase.auth.signInWithPassword({
+    email,
+    password,
+  });
+
+  dispatch({ type: 'SIGN_IN_WITH_EMAIL' });
+  return value;
+};
+
+export const signUp = async (
+  dispatch: AuthDispatch,
+  email: string,
+  password: string,
+) => {
+  dispatch({ type: 'LOADING' });
+  const value = await supabase.auth.signUp({
+    email,
+    password,
+  });
+  dispatch({ type: 'SIGN_UP' });
+
+  return value;
+};
+
+export const signOut = async (dispatch: AuthDispatch) => {
+  dispatch({ type: 'LOADING' });
+  const value = await supabase.auth.signOut();
+  dispatch({ type: 'SIGN_OUT' });
+
+  return value;
+};
+
+export const verifyOtp = async (
+  dispatch: AuthDispatch,
+  email: string,
+  token: string,
+) => {
+  dispatch({ type: 'LOADING' });
+  const value = await supabase.auth.verifyOtp({
+    email,
+    token,
+    type: 'email',
+  });
+  dispatch({ type: 'VERIFY_OTP' });
+
+  return value;
+};
+
+export const resendVerification = async (
+  dispatch: AuthDispatch,
+  email: string,
+) => {
+  dispatch({ type: 'LOADING' });
+  const value = await supabase.auth.resend({
+    type: 'signup',
+    email,
+  });
+  dispatch({ type: 'VERIFY_OTP' });
+
+  return value;
+};
+
+export const resetPassword = async (dispatch: AuthDispatch, email: string) => {
+  dispatch({ type: 'LOADING' });
+  const value = await supabase.auth.resetPasswordForEmail(email);
+  dispatch({ type: 'RESET_PASSWORD' });
+
+  return value;
+};
+
+export const updateUser = async (
```

Comment:
i wasn't able to sign up my own user for some reason, but one thing i was wondering about was that if we needed this function or if the supabase.auth.onAuthStateChange would catch the change triggered by supabase.auth.updateUser. if not then this is fine, but otherwise i think we could remove this function?

Reply:
Did you get an error when you tried to sign up?

Reply:
yea when i clicked the sign up button, i was getting an error of "null value in column "username" of relation "profiles" violates not-null constraint"